src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -27,40 +33,70 @@
     public static final String MESSAGE_INVALID_PREAMBLE = "Incorrect number of arguments in preamble."
             + "The following fields are expected:"
             + "\n";
-    public static final String[] MESSAGE_PREAMBLE_FIELDS = {"property/buyer", "INDEX"};
 
     public static final int ACTOR_POSITIONAL_INDEX = 0;
     public static final int INDEX_POSITIONAL_INDEX = 1;
+    public static final int SORT_TYPE_POSITIONAL_INDEX = 1;
+    public static final int SORT_DIR_POSITIONAL_INDEX = 2;
 
     /**
-     * Parses {@code preamble} into varying fields depending on the specified numFields.
-     * The fields are, in order of index,
-     * 0. Actor (property or buyer)
-     * 1. Index (used by edit etc.)
+     * Parses the preamble, which contains the actor only,
+     * and return a {@code PreambleActorData} that represents
+     * the actor in the {@code preamble}.
      *
-     * @param preamble  Positional arguments in a command.
-     * @param numFields Number of positional arguments expected.
-     * @throws ParseException if the number of positional arguments is different from {@code numFields}
-     *                        or if subsequent parsing of each individual argument throws a ParseException.
+     * @param preamble String that represents an {@code actor}.
+     * @return a {@code PreambleActorData} object.
+     * @throws ParseException if the given string contains the wrong number of keywords or an invalid actor.
      */
-    public static PreambleData parsePreamble(String preamble, int numFields) throws ParseException {
+    public static PreambleActorData parseActorPreamble(String preamble) throws ParseException {
```

Comment:
Instead of needing to create a method and a PreambleXXXData class for each type of command, would it be simpler to have methods of this form that the command parsers can mix and match?
`Actor parseActor(String preamble, int index)`
`SortType parseSortType(String preamble, int index)`

Reply:
can also